auth0/management/roles.py

```diff
@@ -58,7 +58,7 @@ def list(
         per_page: int = 25,
         include_totals: bool = True,
         name_filter: str | None = None,
-    ) -> List[dict[str, Any]]:
+    ):
```

Comment:
You can use `typing.Literal` in combination with `typing.overload` to get dependent types. For example:

```
from typing import Literal, overload

@overload
def list(..., include_totals: Literal[True], ...) -> dict[str, Any]:
    ...

@overload
def list(..., include_totals: Literal[False], ...) -> list[str]:
    ...
```